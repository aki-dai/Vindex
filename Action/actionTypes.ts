export const LOAD_TAG   = 'LOAD_TAG'
export const UPDATE_TAG = 'UPDATE_TAG'
export const DELETE_TAG = 'DELETE_TAG'
export const ADD_TAG    = 'ADD_TAG'

export interface User{
    userID: string
    userName: string
}

export interface Tag{
    user        : User
    value       : string
    timestump   : string
    youtubeID   : string
}


export interface LoadTagAction {
    type: typeof LOAD_TAG
    loadTags    : Tag[]
    youtubeID   : string
}

export interface UpdateTagAction {
    type: typeof UPDATE_TAG
    newTags: Tag[]
}

export interface DeleteTagAction {
    type: typeof DELETE_TAG
    numDeleteTag: number
}

export interface AddTagAction {
    type: typeof ADD_TAG
    newTag: Tag
}

export type TagActionTypes = LoadTagAction   |
                             UpdateTagAction |
                             DeleteTagAction |
                             AddTagAction