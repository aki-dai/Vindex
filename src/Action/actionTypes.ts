export const LOAD_TAG   = 'LOAD_TAG'
export const UPDATE_TAG = 'UPDATE_TAG'
export const DELETE_TAG = 'DELETE_TAG'
export const ADD_TAG    = 'ADD_TAG'

export const FETCH_MOVIE = 'FETCH_MOVIE'
export const ADD_EDIT_TAG = 'ADD_EDIT_TAG'
export const DELETE_EDIT_TAG = 'DELETE_EDIT_TAG'
export const POST_EDIT_TAG = 'POST_EDIT_TAG'

export const AUTH_USER = 'AUTH_USER'
export const UPDATE_ACCESS_TOKEN = 'UPDATE_ACCESS_TOKEN'
export const UPDATE_TOKENS = 'UPDATE_TOKENS'
export const SET_USER_INFO = 'SET_USER_INFO'
export const SIGN_OUT = 'SIGN_OUT'
export const LEAVE_USER = 'LEAVE_USER'

export const SEARCH_SUBMIT = 'SEARCH_ACTION_SUBMIT'
export const SEARCH_START = 'SEARCH_ACTION_START'
export const SEARCH_COMPLETE = 'SEARCH_ACTION_COMPLETE'
export const SEARCH_ERROR = 'SEARCH_ACTION_ERROR'


export type APIStatus = 'initial' | 'waiting' | 'loading' | 'complete' | 'error'

export interface Contributer{
    userID?: string
    userName: string
}

export interface Tag{
    contributer? : Contributer
    count?       : number
    value        : string
    youtubeID    : string
}

export interface User{
    authenticated: boolean
    accessToken: string
    refreshToken: string
    userID: string
    userName: string
    nickName: string
    provider: string
    image: string
}

export type tagTypes = "movie" | "editor"

export interface LoadTagAction {
    type: typeof LOAD_TAG
    loadTags    : Tag[]
    youtubeID   : string
    tagType     : tagTypes
}

export interface UpdateTagAction {
    type: typeof UPDATE_TAG
    newTags: Tag[]
    tagType: tagTypes
}

export interface DeleteTagAction {
    type        : typeof DELETE_TAG
    numDeleteTag: number
    tagType     : tagTypes
}

export interface AddTagAction {
    type    : typeof ADD_TAG
    newTag  : Tag
    tagType : tagTypes
}

export interface FetchMovieAction {
    type        : typeof FETCH_MOVIE
    youtubeID   : string,
    title       : string,
    channelName : string,
    url         : string,
}

export type TagActionTypes = LoadTagAction   |
                             UpdateTagAction |
                             DeleteTagAction |
                             AddTagAction   |
                             FetchMovieAction

export interface AuthUserAction {
    type: typeof AUTH_USER
    accessToken: string
    refreshToken: string
    userID: string
    provider: string
    accessExp: number
}

export interface SetUserInfoAction{
    type: typeof SET_USER_INFO
    userName: string
    nickName: string
    image: string
}

export interface UpdateAccessTokenAction {
    type: typeof UPDATE_ACCESS_TOKEN
    accessToken: string
    accessExp: number
}

export interface UpdateTokensAction {
    type: typeof UPDATE_TOKENS
    accessToken: string
    refreshToken: string
    accessExp: number
}

export interface SignOutUserAction {
    type: typeof SIGN_OUT
}

export interface LeaveUserAction {
    type: typeof LEAVE_USER
}

export type UserActionTypes = AuthUserAction |
                              SetUserInfoAction |
                              UpdateAccessTokenAction|
                              UpdateTokensAction |
                              SignOutUserAction |
                              LeaveUserAction

export interface ResultIndex {
    youtube_id: string
    channelName: string
    title: string
    thumbnail: string
    tags: Tag[]
}

export type sortType = 'latest' | 'oldest' | 'new_movie' | 'previous_movie'

export interface SearchSubmitAction {
    type: typeof SEARCH_SUBMIT
    payload: {
        query: string
        sort: sortType
    }
}
 
export interface SearchStartAction {
    type: typeof SEARCH_START   
}
 
export interface SearchCompleteAction {
    type: typeof SEARCH_COMPLETE
    payload:{
        query: string
        count: number
        sort: sortType
        results:ResultIndex[]
    }
}
 
export interface SearchErrorAction {
    type: typeof SEARCH_ERROR
    query: string
    sort: sortType
    error: unknown
}

export type SearchActionTypes = SearchSubmitAction |
                                SearchStartAction |
                                SearchCompleteAction |
                                SearchErrorAction