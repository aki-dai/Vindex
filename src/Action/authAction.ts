import {AuthAction, AUTH} from './actionTypes'

export function setRedirectUrl(url: string):AuthAction {
    return{
        type: AUTH,
        url: url,
    }
}