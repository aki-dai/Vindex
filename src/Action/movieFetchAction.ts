import {FETCH_MOVIE, FetchMovieAction} from './actionTypes'

export function fetchMovie(vid:string, title:string, channelName: string): FetchMovieAction{
    return {
        type        : FETCH_MOVIE,
        youtubeID   : vid,
        title       : title,
        channelName : channelName,
    }
}
