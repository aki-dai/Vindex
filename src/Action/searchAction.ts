import {SearchActionTypes, SearchIndex, sortType} from './actionTypes'
import { queryToScope } from '../components/functions';

export function SearchSubmit(q: string, sort: sortType): SearchActionTypes{
    return {
        type:'SEARCH_ACTION_SUBMIT',
        payload:{
            query: q,
            sort: sort
        }
    }
}

export function SearchStart(): SearchActionTypes{
    return {
        type:'SEARCH_ACTION_START',
    }
}

export function SearchComplete(q: string, sort: sortType, count:number, results:SearchIndex[]): SearchActionTypes{
    return {
        type:'SEARCH_ACTION_COMPLETE',
        payload:{
            query: q,
            count: count,
            sort: sort,
            results: results
        }
    }
}
 
export function SearchErrorAction(q:string, sort: sortType, error:string) {
    return{
        type: 'SEARCH_ERROR',
        query: q,
        sort: sort,
        error: error,
    } 
}