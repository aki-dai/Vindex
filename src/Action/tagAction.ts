import {Tag, tagTypes, FETCH_MOVIE, LOAD_TAG, UPDATE_TAG, DELETE_TAG, ADD_TAG, TagActionTypes, LOAD_MOVIE_INFO} from './actionTypes'



export function fetchMovie(youtubeID:string, title: string, channelName: string, url: string, tags:Tag[], state: "registered" | "unregistered"): TagActionTypes{
    const isRegistered = (state === "registered")
    return {
        type        : FETCH_MOVIE,
        youtubeID   : youtubeID,
        title       : title,
        channelName : channelName,
        url         : url,
        tags        : tags,
        isRegistered: isRegistered,
    }
}

export function loadMovieInfo(youtubeID:string, channelName: string, title: string): TagActionTypes{
    return {
        type        : LOAD_MOVIE_INFO,
        youtubeID   : youtubeID,
        title       : title,       
        channel_name : channelName,
    }
}

export function loadTag(youtubeID:string, valueTags: Tag[], tagType: tagTypes): TagActionTypes{
    return {
        type        : LOAD_TAG,
        tagType     : tagType,
        loadTags    : valueTags,
        youtubeID   : youtubeID,
    }
}

export function addTag(valueTag: Tag, tagType: tagTypes): TagActionTypes{
    return {
        type    : ADD_TAG,
        newTag  : valueTag,
        tagType : tagType
    }
}

export function deleteTag(numDeleteTag: number, tagType: tagTypes): TagActionTypes{
    console.log({numDeleteTag}, tagType)
    return {
        type            : DELETE_TAG,
        numDeleteTag    : numDeleteTag,
        tagType         : tagType
    }
}

export function updateTag(newTags: Tag[], tagType: tagTypes): TagActionTypes{
    return {
        type    : UPDATE_TAG,
        newTags : newTags,
        tagType : tagType
    }
}