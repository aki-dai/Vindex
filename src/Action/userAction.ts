import {User, AUTH_USER, SET_USER_INFO, UPDATE_TOKEN, SIGN_OUT, LEAVE_USER, UserActionTypes} from './actionTypes'
import { Provider, useDispatch } from 'react-redux';
import Axios from 'axios';

/*
interface UserInfoResponse{
    status: "success" | "failed"
    payload: {
        userName: string
        nickName: string
        image   : string
    }&{
        error: string
    }
}
*/

export function authUser(accessToken:string, refreshToken:string, userID: string, provider: string): UserActionTypes{
   // const dispatch = useDispatch()
   // dispatch(setUserInfo(accessToken))
    return {
        type        : AUTH_USER,
        accessToken : accessToken,
        refreshToken: refreshToken,
        userID      : userID,
        provider    : provider
    }
}

export function setUserInfo(accessToken:string): UserActionTypes{
    let userName =""
    let nickName =""
    let image =""
    console.log({accessToken})
    Axios.get("http://localhost:3000/api/v1/users/",{
        params: {
            access_token: accessToken
        }}).then((res)=>{
            if(res.data.status = "success"){
                console.log({res})
                userName = res.data.payload.userName
                nickName = res.data.payload.nickName
                image = res.data.payload.userName
            }
            else throw new Error(res.data.payload.error)
        }
    )
    return {
        type    : SET_USER_INFO,
        userName: userName,
        nickName: nickName,
        image   : image
    }
}



export function updateToken(accessToken:string, refreshToken:string): UserActionTypes{
    return {
        type        : UPDATE_TOKEN,
        accessToken : accessToken,
        refreshToken: refreshToken,
    }
}

export function signOut(): UserActionTypes{
    return {
        type    : SIGN_OUT,
    }
}

export function leaveUser(): UserActionTypes{

    return {
        type    : LEAVE_USER,
    }
}