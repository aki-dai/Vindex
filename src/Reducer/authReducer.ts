import {AuthAction} from '../Action/actionTypes'

export interface AuthState{
    redirectPath: string
} 

export const initialState: AuthState = {
    redirectPath: ''
}

const authReducer = (state = initialState, action:AuthAction) => {
    switch(action.type) {
        case "AUTH":{
            return{     
                ...state,
                redirectPath: action.url
            }
        }

        default: {
            return state
        }
    }
}

export default authReducer
