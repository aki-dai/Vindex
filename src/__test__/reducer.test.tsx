import authReducer, {initialState} from '../Reducer/authReducer'
import * as actionType from '../Action/actionTypes'

describe('authReducer works collectly',() => {
    it('init',() => {
        expect(authReducer(undefined, {type:'AUTH', url: ''})).toEqual(initialState)
    })
    it('setRedirectPath',() => {
        expect(authReducer(initialState, {type:'AUTH', url: '/about'})).toMatchObject({redirectPath: '/about'})
    })
})
  
