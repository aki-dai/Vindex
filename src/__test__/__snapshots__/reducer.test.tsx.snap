// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userReducer works collectly logOut 1`] = `
Object {
  "initialState": Object {
    "accessExp": null,
    "accessToken": "",
    "authenticated": false,
    "error": "",
    "image": "",
    "nickName": "",
    "provider": "",
    "refreshToken": "",
    "status": "initial",
    "userID": "",
    "userName": "",
  },
}
`;
